common/src/test/java/io/netty/util/internal/NativeLibraryLoaderTest.java

```diff
@@ -51,6 +57,33 @@ public void testFileNotFoundWithNullClassLoader() {
         }
     }
 
+    @Test
+    @EnabledOnOs(LINUX)
+    void testMultipleResourcesInTheClassLoader() throws MalformedURLException {
+        URL url1 = new File("src/test/data/NativeLibraryLoader/1").toURI().toURL();
+        URL url2 = new File("src/test/data/NativeLibraryLoader/2").toURI().toURL();
+        URLClassLoader loader = new URLClassLoader(new URL[] {url1, url2});
+        String resourceName = "test1";
+        try {
+            NativeLibraryLoader.load(resourceName, loader);
+            fail("Expected IllegalStateException because there are more than one resources with the same name");
+        } catch (IllegalStateException ise) {
+            assertTrue(ise.getMessage()
+                    .contains("Multiple resources found for 'META-INF/native/lib" + resourceName + ".so'"));
+        }
+    }
+
+    @Test
+    @EnabledOnOs(LINUX)
```

Comment:
Do we also need to ensure it is only run on `x86_64` ?

Reply:
Done!